src/devices/Card/CreditCard/CreditCard.cs

```diff
@@ -0,0 +1,497 @@
+                        var templates = Tags.Where(m => m.TagNumber == 0x6F).Where(m => m.Tags.Where(t => t.TagNumber == 0x84).Where(t => t.Data.SequenceEqual(appIdentifier.Data)) != null).Where(m => m.Tags.Where(t => t.TagNumber == 0xA5) != null);
+                        // Only here we may find a PDOL tag 0X9F38
+                        Tag pdol = null;
+                        foreach (var temp in templates)
+                        {
+                            // We are sure to have 0xA5, so select it and search for PDOL
+                            pdol = Tag.SearchTag(temp.Tags, 0xA5).FirstOrDefault().Tags.Where(m => m.TagNumber == 0x9F38).FirstOrDefault();
+                            if (pdol != null)
+                                break;
+                        }
+
+                        Span<byte> received = stackalloc byte[260];
+                        byte sumDol = 0;
+                        // Do we have a PDOL?
+                        if (pdol != null)
+                        {
+                            // So we need to send as may bytes as it request
+                            foreach (var dol in pdol.Tags)
+                            {
+                                sumDol += dol.Data[0];
+                            }
+                        }
+
+                        // We send only 0 but the right number
+                        Span<byte> toSend = new byte[2 + sumDol];
+                        // Template command, Tag 83
+                        toSend[0] = 0x83;
+                        toSend[1] = sumDol;
+                        // If we have a PDOL, then we need to fill it properly
+                        // Some fields are mandatory
+                        int index = 2;
+                        if (pdol != null)
+                        {
+                            foreach (var dol in pdol.Tags)
+                            {
+                                // TerminalTransactionQualifier 
+                                if (dol.TagNumber == 0x9F66)
+                                {
+                                    // Select modes to get a maximum of data
+                                    TerminalTransactionQualifier ttq = TerminalTransactionQualifier.MagStripeSupported | TerminalTransactionQualifier.EmvModeSuypported | TerminalTransactionQualifier.EmvContachChipSupported |
+                                        TerminalTransactionQualifier.OnlinePinSupported | TerminalTransactionQualifier.SignatureSupported | TerminalTransactionQualifier.ConstactChipOfflinePinSupported |
+                                        TerminalTransactionQualifier.ConsumerDeviceCvmSupported | TerminalTransactionQualifier.IssuerUpdateProcessingSupported;
+                                    // Encode the TTq
+                                    toSend[index] = (byte)((long)ttq >> 16);
```

Comment:
I think (uint) might be a better choice, also you might want to use `BinaryPrimitives.WriteInt32Little/BigEndian`

Reply:
moved to uint, yes, will be enough. I only have 3 bytes to convert, reason why I didn't use the primitives. So I changed the enum as well to uint so I'm adding the last byte which is always 0 and I'm using the BinaryPrimitives converter then.